src/common/tests_dmg_helpers.c

```diff
@@ -415,12 +415,12 @@ dmg_pool_create(const char *dmg_config_file,
 	if (prop != NULL) {
 		entry = daos_prop_entry_get(prop, DAOS_PROP_PO_ACL);
 		if (entry != NULL) {
-			tmp_name = tmpnam(tmp_buf);
-			if (tmp_name == NULL) {
-				D_ERROR("failed to create tmpfile name\n");
+			fd = mkstemp(tmp_name);
+			if (fd < 0) {
+				D_ERROR("failed to create tmpfile file\n");
 				D_GOTO(out_cmd, rc = -DER_NOMEM);
 			}
-			tmp_file = fopen(tmp_name, "w");
+			tmp_file = fdopen(fd, "w");
```

Comment:
I think this is not needed since mkstemp function creates and opens the file, so the fd should already be the open file descriptor for reading and writing.

Reply:
The `print_acl_entry()` expects a stream (`FILE *`) rather than a file descriptor. That's why I use `fdopen(3)` to associate a stream with the file descriptor already created by `mkstemp(3)`

Reply:
Sounds good then.